service/src/main/scala/za/co/absa/logingw/rest/config/GitConfig.scala

```diff
@@ -18,42 +18,60 @@ package za.co.absa.logingw.rest.config
 
 import org.springframework.boot.context.properties.{ConfigurationProperties, ConstructorBinding}
 import javax.annotation.PostConstruct
-import java.io.PrintWriter
 import java.text.SimpleDateFormat
+import java.io.{PrintWriter, IOException}
+import scala.util.Try
 
 @ConstructorBinding
 @ConfigurationProperties(prefix = "logingw.rest.config")
-case class GitConfig(gitProperties: Boolean) {
+case class GitConfig(generateGitProperties: Boolean) {
 
   @PostConstruct
   def init(): Unit = {
-    if(gitProperties)
-    this.GitPropertiesGenerator.generateGitProperties()
+    if(generateGitProperties)
+      GitPropertiesGenerator.generateGitProperties()
   }
+}
+
+case class GitProperties(branch: String, commitId: String, commitTime: String)
+
+object GitPropertiesHolder {
+  var gitProperties: GitProperties = _
+}
```

Comment:
This looks a unusal -- it does not seem very Scala idiomatic. Is there any other way to do it without the `var`? Perhaps some sort of dependency injection?